src/components/Reactions/ReactionsList.tsx

```diff
@@ -30,113 +29,79 @@ export type ReactionsListProps<
-          return (
-            ReactionOption && (
-              <li
-                className={clsx('str-chat__message-reaction', {
-                  'str-chat__message-reaction-own': isOwnReaction,
-                })}
-                key={reactionType}
-              >
-                <ButtonWithTooltip
-                  aria-label={`Reactions: ${reactionType}`}
-                  data-testid={`reactions-list-button-${reactionType}`}
-                  title={aggregatedUserNamesByType[reactionType].join(', ')}
-                  type='button'
+    <>
+      <div
+        aria-label='Reaction list'
+        className={clsx('str-chat__reaction-list str-chat__message-reactions-container', {
+          'str-chat__reaction-list--reverse': reverse,
+        })}
+        data-testid='reaction-list'
+        role='figure'
+      >
+        <ul className='str-chat__message-reactions'>
+          {existingReactions.map(
+            ({ EmojiComponent, isOwnReaction, reactionCount, reactionType }) =>
+              EmojiComponent && (
+                <li
+                  className={clsx('str-chat__message-reaction', {
+                    'str-chat__message-reaction-own': isOwnReaction,
+                  })}
+                  key={reactionType}
                 >
-                  <span className='str-chat__message-reaction-emoji'>
-                    <ReactionOption.Component />
-                  </span>
-                  &nbsp;
-                  <span
-                    className='str-chat__message-reaction-count'
-                    data-testclass='reaction-list-reaction-count'
+                  <button
+                    aria-label={`Reactions: ${reactionType}`}
+                    data-testid={`reactions-list-button-${reactionType}`}
+                    onClick={() => handleReactionButtonClick(reactionType)}
+                    type='button'
                   >
-                    {reactionCounts[reactionType]}
-                  </span>
-                </ButtonWithTooltip>
-              </li>
-            )
-          );
-        })}
-        <li>
-          <span className='str-chat__reaction-list--counter'>{totalReactionCount}</span>
-        </li>
-      </ul>
-    </div>
+                    <span className='str-chat__message-reaction-emoji'>
+                      <EmojiComponent />
+                    </span>
+                    &nbsp;
+                    <span
+                      className='str-chat__message-reaction-count'
+                      data-testclass='reaction-list-reaction-count'
+                    >
+                      {reactionCount}
+                    </span>
+                  </button>
+                </li>
+              ),
+          )}
+          <li>
+            <span className='str-chat__reaction-list--counter'>{totalReactionCount}</span>
+          </li>
+        </ul>
+      </div>
+      <ReactionsListModal
```

Comment:
🍓

Reply:
Would it make sense to make this customizable through component context? I am not sure as ReactionsList and ReactionsListModal do not contain complicated logic.